### PR TITLE
StrictFakeOptions.Allow* do not apply to overridden Object methods

### PR DESCRIPTION
**Pull request by blairconrad**

### CODE REVIEW

**Comment by blairconrad** [comment]
Currently builds on #1849. I can rebase once that's merged.

File: tests/FakeItEasy.Specs/StrictFakeSpecs.cs [hunk 1]
```diff
@@ -58,6 +58,26 @@ public interface IMyInterface
                 .x(() => exception.Should().BeAnExceptionOfType<ExpectationException>());
         }
```
Comment (blairconrad) [inline tests/FakeItEasy.Specs/StrictFakeSpecs.cs:60]: None of the new tests initially passed.

**Review by thomaslevesque — APPROVED** [review]
Looks fine to me. And it looks like I had made the same mistake again with IsSameMethodAs/HasSameBaseMethodAs. Thanks @blairconrad!

**Comment by thomaslevesque** [comment]
Should be merged automatically after you rebase

**Comment by blairconrad** [comment]
Whoops! I foolishly changed the title before I rebased. Why? I don't know. I was here and not thinking and it automerged. We got an ugly graph. Sorry about that. Also, thanks for the review and auto-merge!

**Pull request merged**